zaza/openstack/charm_tests/designate/tests.py

```diff
@@ -194,4 +250,5 @@ def test_400_domain_creation(self):
 
         logging.debug('Tidy up delete test record')
         self.zones_delete(_domain_id)
+        self._wait_on_domain_gone()
```

Comment:
I think this will wait on the wrong domain_id?  I meant a function that waits on `_domain_id`?  However, I don't now as it's very confusing what is going on.  Does this wait on the right domain id?

Reply:
@ajkavanagh , i think i have refactored this in a way that is more straight forward... let me know if it doesn't